src/design-system/components/MarkdownText/MarkdownText.tsx

```diff
@@ -0,0 +1,315 @@
+import React, {
+  Children,
+  createContext,
+  Fragment,
+  memo,
+  ReactNode,
+  useContext,
+  useMemo,
+} from 'react';
+import { Text as NativeText, StyleSheet, View } from 'react-native';
+import MarkdownDisplay, {
+  ASTNode,
+  RenderRules,
+} from 'react-native-markdown-display';
+import { negateSpace, Space } from '../../layout/space';
+import { renderStringWithEmoji } from '../../typography/renderStringWithEmoji';
+import { fonts } from '../../typography/typography';
+import { Box } from '../Box/Box';
+import { Text, TextProps } from '../Text/Text';
+import { TextLink } from '../TextLink/TextLink';
+
+const styles = StyleSheet.create({
+  code: { ...fonts.SFMono.medium, letterSpacing: 0 },
+  strong: fonts.SFProRounded.semibold,
+  tableCell: { flex: 1 },
+  tabularNumbers: { fontVariant: ['tabular-nums'] },
+});
+
+const defaultProps: {
+  nestedSpace: Space;
+  space: Space;
+  size: NonNullable<TextProps['size']>;
+} = {
+  nestedSpace: '19px',
+  size: '16px',
+  space: '30px',
+};
+
+interface MarkdownStackContextObject {
+  space: Space;
+  nestedSpace: Space;
+  depth: number;
+}
+const MarkdownStackContext = createContext<MarkdownStackContextObject>({
+  depth: 0,
+  nestedSpace: defaultProps.nestedSpace,
+  space: defaultProps.space,
+});
+
+interface MarkdownStackProps {
+  space?: Space;
+  nestedSpace?: Space;
+  children: ReactNode;
+}
+
+function MarkdownStack({
+  space = defaultProps.space,
+  nestedSpace = defaultProps.nestedSpace,
+  children,
+}: MarkdownStackProps) {
+  const depth = useContext(MarkdownStackContext).depth + 1;
+
+  return (
+    <MarkdownStackContext.Provider
+      value={useMemo(() => ({ depth, nestedSpace, space }), [
+        depth,
+        space,
+        nestedSpace,
+      ])}
+    >
+      <Box marginBottom={negateSpace(depth === 1 ? space : nestedSpace)}>
+        {children}
+      </Box>
+    </MarkdownStackContext.Provider>
+  );
+}
+
+interface MarkdownStackItemProps {
+  children: ReactNode;
+}
+
+function MarkdownStackItem({ children }: MarkdownStackItemProps) {
+  const { depth, space, nestedSpace } = useContext(MarkdownStackContext);
+
+  return (
+    <Box paddingBottom={depth === 1 ? space : nestedSpace}>{children}</Box>
+  );
+}
+
+function renderBullet(parents: ASTNode[], index: number): ReactNode {
+  const orderedListIndex = parents.findIndex(el => el.type === 'ordered_list');
+
+  if (orderedListIndex > -1) {
+    const orderedList = parents[orderedListIndex];
+    const listItemNumber = orderedList.attributes?.start
+      ? orderedList.attributes.start + index
+      : index + 1;
+
+    return (
+      <NativeText style={styles.tabularNumbers}>{listItemNumber}.</NativeText>
+    );
+  }
+
+  return `${ios ? '\u00B7' : '\u2022'}`;
```

Comment:
Do you have an opinion that we should eventually release this as a separate package? If yes, maybe we should,'t use this helper. 
At the same time, changing this later is 5 minutes

Reply:
Good catch. Moving this into its own repo is definitely something I'm thinking about long term, so I've been trying to keep the `design-system` folder decoupled from the rest of the app. I'll fix this up.